### ISSUE TITLE
[Select] listbox should indicate the focused option item

### ISSUE THREAD

**Issue by noahchoii** [mui/base-ui#2375]
# Bug report

## Current behavior

There's no `aria-activedescendant` on `role="listbox"` element.

<img width="1125" height="392" alt="Image" src="https://github.com/user-attachments/assets/88f9b984-83d1-44d5-bb1d-e0dbdc380122" />

## Expected behavior

`role="listbox"` element should have `aria-active-descendant` attribute. It refers the id of focused `role="option"` element.

[This](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/listbox_role#states_and_properties
) is MDN explanation about these a11y.

## Reproducible example

Link to a CodeSandbox, StackBlitz, or other IDE.

## Base UI version

v1.0.0-beta.1

## Which browser are you using?

Chrome

## Which OS are you using?

Mac OS

## Which assistive tech are you using (if applicable)?

Voiceover

## Additional context

-
